File: ARCADE/The Core/Level 4_10: countBlackCells.cs
    using System;
     
    public class Test
    {
    	public static int countBlackCells(int n, int m){
            return n + m - 2 + gcd(n, m);
        }
     
        public static int gcd(int a, int b){
            return a == 0 ? b : gcd(b % a, a);
        }
    	public static void Main()
    	{
    		int n = 3, m = 4;
    		Console.WriteLine(countBlackCells(n, m));
    		int p = 3, q = 3;
    		Console.WriteLine(countBlackCells(p, q));
    	}
    }

File: ARCADE/The Core/Level 4_10: countBlackCells.pas
program countBlackCells;

function gcd(a: integer; b: integer): integer;
begin
    if(a = 0) then gcd := b
    else gcd := gcd(b mod a, a);
end;

function countBlackCells(n: integer; m: integer): integer;
begin
    countBlackCells := n + m - 2 + gcd(n, m);
end;

var n, m : integer;
begin
	n := 3; m := 4;
	writeln(countBlackCells(n, m));
	n := 3; m := 3;
	writeln(countBlackCells(n, m));
end.
